File: backend/asp/src/main/java/com/caps/asp/model/uimodel/response/common/UserResponseModel.java
package com.caps.asp.model.uimodel.response.common;

import com.caps.asp.model.TbUser;
import com.caps.asp.model.TbUserRate;
import com.caps.asp.model.uimodel.common.UserSuggestSettingModel;
import org.springframework.format.annotation.DateTimeFormat;

import java.sql.Date;
import java.util.List;

public class UserResponseModel {
    private Integer userId;
    private String fullname;
    private String imageProfile;
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date dob;
    private String phone;
    private int gender;
    private int roleId;
    private String email;
    private String username;
    private Double avgBehaviourRate;
    private Double avgLifeStyleRate;
    private Double avgPaymentRate;
    private List<UserRateResponseModel> userRateResponseModels;
    private UserSuggestSettingModel userSuggestSettingModel;

    public UserResponseModel() {
    }
    public UserResponseModel(TbUser user){
        this.userId = user.getUserId();
        this.fullname = user.getFullname();
        this.imageProfile = user.getImageProfile();
        this.dob = user.getDob();
        this.phone = user.getPhone();
        this.gender = user.getGender();
        this.email = user.getEmail();
        this.roleId = user.getRoleId();
        this.username = user.getUsername();
    }

    public UserResponseModel(Integer userId, String fullname, String imageProfile, Date dob, String phone, int gender, int roleId, String email, String username, Double avgBehaviourRate, Double avgLifeStyleRate, Double avgPaymentRate, List<UserRateResponseModel> userRateResponseModels, UserSuggestSettingModel userSuggestSettingModel) {
        this.userId = userId;
        this.fullname = fullname;
        this.imageProfile = imageProfile;
        this.dob = dob;
        this.phone = phone;
        this.gender = gender;
        this.roleId = roleId;
        this.email = email;
        this.username = username;
        this.avgBehaviourRate = avgBehaviourRate;
        this.avgLifeStyleRate = avgLifeStyleRate;
        this.avgPaymentRate = avgPaymentRate;
        this.userRateResponseModels = userRateResponseModels;
        this.userSuggestSettingModel = userSuggestSettingModel;
    }

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public String getFullname() {
        return fullname;
    }

    public void setFullname(String fullname) {
        this.fullname = fullname;
    }

    public String getImageProfile() {
        return imageProfile;
    }

    public void setImageProfile(String imageProfile) {
        this.imageProfile = imageProfile;
    }

    public Date getDob() {
        return dob;
    }

    public void setDob(Date dob) {
        this.dob = dob;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public int getGender() {
        return gender;
    }

    public void setGender(int gender) {
        this.gender = gender;
    }

    public int getRoleId() {
        return roleId;
    }

    public void setRoleId(int roleId) {
        this.roleId = roleId;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public Double getAvgBehaviourRate() {
        return avgBehaviourRate;
    }

    public void setAvgBehaviourRate(Double avgBehaviourRate) {
        this.avgBehaviourRate = avgBehaviourRate;
    }

    public Double getAvgLifeStyleRate() {
        return avgLifeStyleRate;
    }

    public void setAvgLifeStyleRate(Double avgLifeStyleRate) {
        this.avgLifeStyleRate = avgLifeStyleRate;
    }

    public Double getAvgPaymentRate() {
        return avgPaymentRate;
    }

    public void setAvgPaymentRate(Double avgPaymentRate) {
        this.avgPaymentRate = avgPaymentRate;
    }

    public List<UserRateResponseModel> getUserRateResponseModels() {
        return userRateResponseModels;
    }

    public void setUserRateResponseModels(List<UserRateResponseModel> userRateResponseModels) {
        this.userRateResponseModels = userRateResponseModels;
    }

    public UserSuggestSettingModel getUserSuggestSettingModel() {
        return userSuggestSettingModel;
    }

    public void setUserSuggestSettingModel(UserSuggestSettingModel userSuggestSettingModel) {
        this.userSuggestSettingModel = userSuggestSettingModel;
    }
}

File: IOS-UI/Roommate/Model/API/Response/UserResponseModel.swift
//
//  UserResponeModel.swift
//  Roommate
//
//  Created by TrinhHC on 10/17/18.
//  Copyright © 2018 TrinhHC. All rights reserved.
//
import Foundation
import ObjectMapper
class UserResponseModel: Mappable {
    var userId = 0
    var username:String?
    var email:String?
    var fullname:String?
    var imageProfile:String?
    var dob:Date?
    var phone:String?
    var gender = 1
    var roleId = 4
    var avgBehaviourRate = 0.0
    var avgLifeStyleRate = 0.0
    var avgPaymentRate = 0.0
    var userRateResponseModels:[UserRateResponseModel]?
    var suggestSettingMappedModel:SuggestSettingMappableModel?
    
    init() {
        
    }
    
    public init(userModel:UserModel){
        self.userId = userModel.userId
        self.username = userModel.username
        self.roleId = userModel.roleId
        self.email = userModel.email
        self.fullname = userModel.fullname
        self.imageProfile = userModel.imageProfile
        self.dob = userModel.dob
        self.phone = userModel.phone
        self.gender = userModel.gender
        if let suggestSettingModel  = userModel.suggestSettingModel{
            self.suggestSettingMappedModel = SuggestSettingMappableModel(suggestSettingModel: suggestSettingModel)
        }
        
    }
    public init(userId: Int = 0, username: String?, email: String?, fullname: String?, imageProfile: String?, dob: Date?, phone: String?, gender: Int, userRateResponseModels:[UserRateResponseModel]?,suggestSettingMappedModel: SuggestSettingMappableModel?) {
        self.userId = userId
        self.username = username
        self.email = email
        self.fullname = fullname
        self.imageProfile = imageProfile
        self.dob = dob
        self.phone = phone
        self.gender = gender
        self.userRateResponseModels = userRateResponseModels
        self.suggestSettingMappedModel = suggestSettingMappedModel
    }
    
    
    //MARK: ObjectMapper
    required init?(map: Map) {
    }
    
    func mapping(map: Map){
        userId <- map["userId"]
        username <- map["username"]
        email <- map["email"]
        fullname <- map["fullname"]
        imageProfile <- map["imageProfile"]
        dob <- (map["dob"],CustomDateFormatTransform(formatString: Constants.USER_DATE_FORMAT))
        phone <- map["phone"]
        gender <- map["gender"]
        roleId <- map["roleId"]
        avgBehaviourRate <- map["avgBehaviourRate"]
        avgLifeStyleRate <- map["avgLifeStyleRate"]
        avgPaymentRate <- map["avgPaymentRate"]
        userRateResponseModels <- map["userRateResponseModels"]
        suggestSettingMappedModel <- map["userSuggestSettingModel"]
    }
}
